pkg/controller/infrastructure/actuator.go

```diff
@@ -79,6 +80,7 @@ func NewActuatorWithDeps(
 	chartRendererFactory extensionschartrenderer.Factory,
 	terraformChartOps TerraformChartOps,
 	machineImageOwnerSecretRef *corev1.SecretReference,
+	whitelistedImageIDs *[]string,
```

Comment:
why using one pointer of slice?

Reply:
In case user does not provide a whitelist, `whitelistedImageIDs` would be nil. Then we can use `if whitelistedImageIDs == nil` to determine whether or not they are configured.

Reply:
As I know, we can directly use `nil` to determine if the one slice is nil or not. So why not use `whitelistedImageIDs == nil` or `len(whitelistedImageIDs) == 0` to determine if the slice is nil or empty?

Reply:
will remove slice pointer